core/src/main/java/io/kestra/core/http/client/HttpClient.java

```diff
@@ -0,0 +1,234 @@
+        if (this.client != null) {
+            return this.client;
+        }
+
+        org.apache.hc.client5.http.impl.classic.HttpClientBuilder builder = HttpClients.custom()
+            .disableDefaultUserAgent()
+            .setUserAgent("Kestra");
+
+        // logger
+        if (runContext.logger() != null) {
+            if (ArrayUtils.contains(this.configuration.getLogs(), HttpConfiguration.LoggingType.REQUEST_HEADERS) ||
+                ArrayUtils.contains(this.configuration.getLogs(), HttpConfiguration.LoggingType.REQUEST_BODY)
+            ) {
+                builder.addRequestInterceptorLast(new LoggingRequestInterceptor(runContext.logger(), this.configuration.getLogs()));
+            }
+
+            if (ArrayUtils.contains(this.configuration.getLogs(), HttpConfiguration.LoggingType.RESPONSE_HEADERS) ||
+                ArrayUtils.contains(this.configuration.getLogs(), HttpConfiguration.LoggingType.RESPONSE_BODY)
+            ) {
+                builder.addResponseInterceptorLast(new LoggingResponseInterceptor(runContext.logger(), this.configuration.getLogs()));
+            }
+        }
+
+        // Object dependencies
+        PoolingHttpClientConnectionManagerBuilder connectionManagerBuilder = PoolingHttpClientConnectionManagerBuilder.create();
+        ConnectionConfig.Builder connectionConfig = ConnectionConfig.custom();
+
+        // Timeout
+        if (this.configuration.getTimeout() != null) {
+            if (this.configuration.getTimeout().getConnectTimeout() != null) {
+                connectionConfig.setConnectTimeout(Timeout.of(this.configuration.getTimeout().getConnectTimeout()));
+            }
+
+            if (this.configuration.getTimeout().getReadIdleTimeout() != null) {
+                connectionConfig.setSocketTimeout(Timeout.of(this.configuration.getTimeout().getReadIdleTimeout()));
+            }
+        }
+
+        // proxy
+        if (this.configuration.getProxy() != null) {
+            // @TODO use CustomSocketFactory
+
+            if (this.configuration.getProxy().getUsername() != null && this.configuration.getProxy().getPassword() != null) {
+                builder.setProxyAuthenticationStrategy(new DefaultAuthenticationStrategy());
+
+                BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                credentialsProvider.setCredentials(
+                    new AuthScope(
+                        runContext.render(this.configuration.getProxy().getAddress()),
+                        this.configuration.getProxy().getPort()
+                    ),
+                    new UsernamePasswordCredentials(
+                        runContext.render(this.configuration.getProxy().getUsername()),
+                        runContext.render(this.configuration.getProxy().getPassword()).toCharArray()
+                    )
+                );
+
+                builder.setDefaultCredentialsProvider(credentialsProvider);
```

Comment:
This seems odd, you create a credential provider for the proxy and set it as the default?

Also, it didn't support basic/token auth?